### ISSUE TITLE
Ad notifications fail to open if Brave is closed after the Ad notification is shown

### ISSUE THREAD

**Issue by tmancey** [brave/browser-android-tabs#1616]
## Ad notifications fail to open if Brave is closed after the Ad notification is shown 
Issue also affects macOS, Windows and Linux desktop applications

## Steps to reproduce 

   1. Enable Ads
   2. Visit www.brave.com
   3. Wait for an Ad notification to appear (see Additional Information on how to reduce the wait time for an Ad to appear)
   4. Quit Brave
   5. Tap the Ad notification

or 

   1. Enable Ads
   2. Visit www.brave.com
   3. Wait for an Ad notification to appear (see Additional Information on how to reduce the wait time for an Ad to appear)
   4. Quit Brave
   4. Launch Brave
   5. Tap the Ad notification

## Actual result 
Ad is not opened in a new tab

## Expected result
Ad should be opened in a new tab

## Issue reproduces how often 
Easily reproduced

## Issue happens on 
   - Current Play Store version? No
   - Beta build? No

## Device details
   - Install type (ARM, x86): ARM
   - Device (Phone, Tablet, Phablet): Pixel3 Phone
   - Android version: 9
   
## Brave version
Brave Ads

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Is the issue reproducible on the latest version of Chrome? 

### Additional information
You can reduce the time it takes for an Ad notification to appear by temporarily changing:
```
uint64_t start_timer_in =
    base::Time::kSecondsPerHour / ads_client_->GetAdsPerHour();
```

to

```
uint64_t start_timer_in = 30;  // Ad notification will be shown every 30 seconds
```

in `AdsImpl::StartDeliveringNotifications`

**Comment by tmancey** [comment]
@anthonypkeane @mandar-brave This issue will be resolved by https://github.com/brave/brave-browser/issues/4102

**Comment by lukemulks** [comment]
I have some additional info that may prove helpful. 

Brave v1.1.0, Chromium 75.0.3770.67 

When I initially installed this build ^, I was experiencing the expected clickthrough behavior for the notifications. I'd tap on the notification, and the new tab would open in Brave with the landing page. 

The previous version I was testing was not behaving this way. Clickthroughs were not resulting in the landing page opening in Brave. 

- Yesterday, in the version noted above, I ran into the same issue again with the ads not clicking through. - Initially, clickthroughs behaved as expected (tested during regular business hours PST) 
- I had not quit out of Brave, but had switched from Brave to other apps on my device. 
- I observed the clickthroughs not working as expected around 7:48pm PST. 

However, this morning, I tested clickthrough behavior again, and it is now clicking through as expected. 
- Tested slightly before 9am PST. Clickthrough works as expected.
- Just tested again at 10:45am PST. Clickthrough works as expected. 
- I did not quit Brave between yesterday (when clickthroughs were not working as expected) and today (when clickthroughs were behaving as expected)

Two thoughts: 
1. The mix of expected and unexpected behavior, with the timing that the behavior is occurring, has me wondering if the issue may potentially be occurring during after "work" hours. If I recall correctly, there's logic in place that does not display ads when Brave is backgrounded during specific hours of the day.

2. After clickthroughs were not working as expected last night, I had cleared all notifications from the tray. I did not receive an ad after that took place, so cannot confirm with confidence if clearing the notifications potentially resolved the issue. It may be worth checking to see if when the issue is observed, if clearing the notifications from the tray resolves the issue. If I can reproduce the issue again today, I'll clear notifications and attempt to test, but may be worth testing as well in case I'm unable to repro. 

Hope this helps. Happy to provide additional info as needed.
